hasher-matcher-actioner/hmalib/matchers/filters.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+from sys import maxsize
+import typing as t
+import time
+import functools
+import methodtools
+
+from threatexchange.signal_type.index import IndexMatch
+from threatexchange.signal_type.pdq import PdqSignal
+from threatexchange.signal_type.signal_base import SignalType
+
+from hmalib.indexers.metadata import (
+    BANKS_SOURCE_SHORT_CODE,
+    THREAT_EXCHANGE_SOURCE_SHORT_CODE,
+    BankedSignalIndexMetadata,
+    BaseIndexMetadata,
+    ThreatExchangeIndicatorIndexMetadata,
+)
+from hmalib.common.logging import get_logger
+from hmalib.common.configs.fetcher import (
+    ThreatExchangeConfig,
+    AdditionalMatchSettingsConfig,
+)
+from hmalib.common.models.bank import BankMember, BanksTable
+
+
+logger = get_logger(__name__)
+
+
+PG_CONFIG_CACHE_TIME_SECONDS = 300
+BANK_CACHE_TIME_SECONDS = 300
+BANK_CACHE_SIZE = 200
+
+
+class BaseMatchFilter:
+    """
+    All classes implementing their own match filter logic should implement this
+    interface.
+
+    A sub class can either (re)define `filter_matches` or just
+    `should_filter_out` (RECOMMENDED). The benefit of defining just
+    `should_filter_out` is that the logging is centrally handled and we reduce
+    *some* redundancies.
+
+    If you need more control, or there are benefits to handling all matches
+    together, (re)define `filter_matches`.
+    """
+
+    def filter_matches(
```

Comment:
nit: I think this method could use a few more comments, the distinction of `match` vs `BaseIndexMetadata` is pretty tricky

Reply:
Added more comments to `filter_matches` docstring.